src/compiler/program.ts

```diff
@@ -462,7 +462,7 @@ export function createCompilerHostWorker(options: CompilerOptions, setParentNode
         return getDirectoryPath(normalizePath(system.getExecutingFilePath()));
     }
 
-    const newLine = getNewLineCharacter(options, () => system.newLine);
+    const newLine = getNewLineCharacter(options);

```

Comment:
I think you also need it in `createCompilerHostFromProgramHost`
Also if this doesnt have matcing changes in services.ts, we might have issues with `compileOnSave`?